### PR TITLE
Session management and log out API implementation

### PR DESCRIPTION
**Pull request by tabbasum-rean**

### CODE REVIEW

File: src/api/controllers/patient/patient.controller.ts [hunk 1]
```diff
@@ -224,6 +224,12 @@ export class PatientController extends BaseUserController {
             if (!deleted) {
                 throw new ApiError(400, 'User cannot be deleted.');
             }
+            // invalidate all sessions
+            var invalidated = await this._userService.invalidateSession(request.currentUser.SessionId);
```
Comment (kiran-rean) [inline src/api/controllers/patient/patient.controller.ts:228]: Why you are sending session Id for this?
It should invalidate all active sessions for that user.

File: src/api/validators/user/user.validator.ts [hunk 2]
```diff
@@ -24,6 +24,21 @@ export class UserValidator {
         return request.params.id;
     };
 
+    static logoutToken = async (request: express.Request): Promise<string> => {
```
Comment (kiran-rean) [inline src/api/validators/user/user.validator.ts:27]: Why this token is sent in the request body?
When the user is already logged in, the token is part of authorization header.

File: src/api/controllers/patient/patient.controller.ts (hunk 1 above)
Comment (kiran-rean) [reply src/api/controllers/patient/patient.controller.ts:228]: Create a separate method which expires all the active sessions.
This should be separate from the method which expires a single session during logout.

**Review by kiran-rean — CHANGES_REQUESTED** [review]

File: src/api/validators/user/user.validator.ts (hunk 2 above)
Comment (tabbasum-rean) [reply src/api/validators/user/user.validator.ts:27]: This is not jwt token, it is actually firebase token stored in `user_device_details`. We need to destroy this token when user logs out from that device.

File: src/api/controllers/patient/patient.controller.ts (hunk 1 above)
Comment (tabbasum-rean) [reply src/api/controllers/patient/patient.controller.ts:228]: Understood. I will update this

**Pull request merged**